tests/api/v1/samples/upsert.js

```diff
@@ -63,6 +63,26 @@ describe(`api: POST ${path}`, () => {
   afterEach(u.forceDelete);
   after(tu.forceDeleteUser);
 
+  it('name field is required', (done) => {
```

Comment:
Can you add a couple more tests here (and for bulk upsert, too):
- upsert a sample whose name refers to an aspect which doesn't exist
- upsert a sample whose name refers to an aspect which is not published
- upsert a sample whose name refers to a subject which doesn't exist
- upsert a sample whose name refers to a subject which is not published